core/chains/evm/txmgr/orm_test.go

```diff
@@ -172,3 +172,47 @@ func TestORM(t *testing.T) {
 		assert.Equal(t, r.BlockHash, etx.EthTxAttempts[0].EthReceipts[0].BlockHash)
 	})
 }
+
+func TestORM_FindEthTxAttemptConfirmedByEthTxIDs(t *testing.T) {
+	db := pgtest.NewSqlxDB(t)
+	cfg := configtest.NewGeneralConfig(t, nil)
+	orm := cltest.NewTxmORM(t, db, cfg)
+	ethKeyStore := cltest.NewKeyStore(t, db, cfg).Eth()
+
+	_, from := cltest.MustInsertRandomKey(t, ethKeyStore, 0)
+
+	tx1 := cltest.MustInsertConfirmedEthTxWithLegacyAttempt(t, orm, 0, 1, from) // tx1
+	tx2 := cltest.MustInsertConfirmedEthTxWithLegacyAttempt(t, orm, 1, 2, from) // tx2
```

Comment:
Let's make sure it ignores unconfirmed ones and additional attempts that have no receipts as well

Reply:
I assumed the below assertions cover those conditions, please correct me if I'm missing something.
https://github.com/smartcontractkit/chainlink/blob/ab60f8e46b4b5ae10d69bbc7abbc33d7d2606dc4/core/chains/evm/txmgr/orm_test.go#L212-L217

Reply:
They do not cover it. You only have confirmed etx in the DB and each one has exactly one attempt, with a receipt. I'd like to see some more variety here, e.g. including unconfirmed eth_txes as well and adding additional, unconfirmed attempts to the confirmed etcs.

Reply:
That should be here
https://github.com/smartcontractkit/chainlink/blob/ab60f8e46b4b5ae10d69bbc7abbc33d7d2606dc4/core/chains/evm/txmgr/orm_test.go#L188-L203

Reply:
So 3 transactions
 tx1 - with one attempt
 tx2 - 2 attempts, 1 with receipt
 tx3 - no attempt
 Perhaps ill add unconfirmed transactions too.

Reply:
gotcha, 
attempt2 of tx2 is the one with the receipt already. so I guess we are good here?